projects/sds/pkg/server/server.go

```diff
@@ -0,0 +1,136 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"hash/fnv"
+	"io/ioutil"
+	"net"
+	"os"
+
+	"github.com/solo-io/go-utils/hashutils"
+
+	"github.com/solo-io/go-utils/contextutils"
+
+	"google.golang.org/grpc"
+
+	auth "github.com/envoyproxy/go-control-plane/envoy/api/v2/auth"
+	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	sds "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v2"
+	"github.com/envoyproxy/go-control-plane/pkg/cache"
+	"github.com/envoyproxy/go-control-plane/pkg/server"
+)
+
+const (
+	sdsClient = "sds_client"
+)
+
+var (
+	grpcOptions = []grpc.ServerOption{grpc.MaxConcurrentStreams(10000)}
+)
+
+type EnvoyKey struct{}
+
+func (h *EnvoyKey) ID(node *core.Node) string {
+	return sdsClient
+}
+
+func SetupEnvoySDS() (*grpc.Server, cache.SnapshotCache) {
+	grpcServer := grpc.NewServer(grpcOptions...)
+	hasher := &EnvoyKey{}
+	snapshotCache := cache.NewSnapshotCache(false, hasher, nil)
+	svr := server.NewServer(context.Background(), snapshotCache, nil)
+
+	// register services
+	sds.RegisterSecretDiscoveryServiceServer(grpcServer, svr)
+	return grpcServer, snapshotCache
+}
+
+func RunSDSServer(ctx context.Context, grpcServer *grpc.Server, serverAddress string) error {
+	lis, err := net.Listen("tcp", serverAddress)
+	if err != nil {
+		return err
+	}
+	contextutils.LoggerFrom(ctx).Info(fmt.Sprintf("sds server listening on %s\n", serverAddress))
```

Comment:
FYI (applies to most log entries here): no need to use `Sprintf`, our logger provides some nice features for formatting the entries. E.g. here you can just do:
 ```suggestion
	contextutils.LoggerFrom(ctx).Infof("sds server listening on %s", serverAddress)
```
In general you never need a `/n` at the end of a line with the logger.